spec/spec_helper.rb

```diff
@@ -6,6 +6,9 @@
 require 'shoulda/matchers'
 require 'webmock/rspec'
 
+include Warden::Test::Helpers
+Warden.test_mode!
```

Comment:
Warden is Devise's authentication server that it runs parallel to the rails app.  I believe you have to include this in order to do `login_as(user, :scope => :user)` rather than create a whole login macro.